tests/integration/query/simple/with_filter/with_alias_test.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package simple
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestQuerySimple_WithAliasEqualsFilterBlock_Succeeds(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple query with alias filter(age)",
+		Actions: []any{
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "John",
+					"Age": 21
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "Bob",
+					"Age": 32
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {_alias: {UserAge: {_eq: 21}}}) {
+						Name
+						UserAge: Age
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"Name":    "John",
+							"UserAge": int64(21),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+func TestQuerySimple_WithEmptyAlias_Succeeds(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple query with empty alias filter",
+		Actions: []any{
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "John",
+					"Age": 21
+				}`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"Name": "Bob",
+					"Age": 32
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users(filter: {_alias: {}}) {
+						Name
+						Age
+					}
+				}`,
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"Name": "John",
+							"Age":  int64(21),
+						},
+						{
+							"Name": "Bob",
+							"Age":  int64(32),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	executeTestCase(t, test)
+}
+
+func TestQuerySimple_WithNullAlias_Succeeds(t *testing.T) {
```

Comment:
suggestion: you have a lot of expected result described as "Succeeds". This is not very descriptive. Would be more readable to explain more specifically the expected behaviour. For example in this case you don't expect filter to match any element, then you can write:
`TestQuerySimple_WithNullAlias_FilterShouldNotMatch`.
The same applied to other occurrences.

Reply:
I've updated all of the added tests